frontera/contrib/backends/cassandra/components.py

```diff
@@ -0,0 +1,359 @@
+        db_page.url = obj.url
+        db_page.created_at = datetime.utcnow()
+        db_page.meta = obj.meta
+        if hasattr(obj, 'depth'):
+            db_page.depth = obj.depth
+        else:
+            db_page.depth = 0
+
+        if isinstance(obj, Request):
+            db_page.headers = obj.headers
+            db_page.method = obj.method
+            db_page.cookies = obj.cookies
+        elif isinstance(obj, Response):
+            db_page.headers = obj.request.headers
+            db_page.method = obj.request.method
+            db_page.cookies = obj.request.cookies
+            db_page.status_code = obj.status_code
+        return db_page
+
+    @retry_and_rollback
+    def update_score(self, batch):
+        query = self.session.prepare("UPDATE metadata SET score = ? WHERE fingerprint = ?")
+        cql_items = []
+        for fprint, score, request, schedule in batch:
+            cql_i = (score, fprint)
+            cql_items.append(cql_i)
+        execute_concurrent_with_args(self.session, query, cql_items, concurrency=400)
+        self.cass_count({"scored_urls": len(cql_items)})
+
+    def cass_count(self, counts):
+        for row, count in counts.iteritems():
+            count_page = self.session.prepare("UPDATE crawlstats SET "+row+" = "+row+" + ? WHERE crawl= ?")
+            self.session.execute_async(count_page, (count, self.crawl_id))
+
+
+class States(MemoryStates):
+
+    def __init__(self, session_cls, model_cls, cache_size_limit, crawl_id='default'):
+        super(States, self).__init__(cache_size_limit)
+        self.session = session_cls
+        self.model = model_cls
+        self.table = 'StateModel'
+        self.logger = logging.getLogger("frontera.contrib.backends.cassandra.components.States")
+        self.crawl_id = crawl_id
+
+    @retry_and_rollback
+    def frontier_stop(self):
+        pass
+
+    @retry_and_rollback
+    def fetch(self, fingerprints):
+        to_fetch = [f for f in fingerprints if f not in self._cache]
+        self.logger.debug("cache size %s", len(self._cache))
+        self.logger.debug("to fetch %d from %d", (len(to_fetch), len(fingerprints)))
+
+        for chunk in chunks(to_fetch, 128):
+            for state in self.model.objects.filter(crawl=self.crawl_id, fingerprint__in=chunk):
+                self._cache[state.fingerprint] = state.state
+
+    @retry_and_rollback
+    def flush(self, force_clear=False):
+        query = self.session.prepare("INSERT INTO states (crawl, fingerprint, state) VALUES (?, ?, ?)")
+        cql_items = []
+        for fingerprint, state_val in self._cache.iteritems():
+            cql_i = (self.crawl_id, fingerprint, state_val)
+            cql_items.append(cql_i)
+        execute_concurrent_with_args(self.session, query, cql_items, concurrency=20000)
+        super(States, self).flush(force_clear)
+
+
+class Queue(BaseQueue):
+    def __init__(self, session_cls, queue_cls, partitions, ordering='default', crawl_id='default'):
```

Comment:
There is no need for `default` for `crawl_id` in args, because we already set default value in default_settings.py module.